packages/teko/src/Teko_Utilities.cpp

```diff
@@ -1833,6 +1841,34 @@ const LinearOp explicitAdd(const LinearOp & opl_in,const LinearOp & opr_in)
    bool isTpetral = Teko::TpetraHelpers::isTpetraLinearOp(opl);
    bool isTpetrar = Teko::TpetraHelpers::isTpetraLinearOp(opr);
 
+   // if one of the operators in the sum is a thyra zero op
+   if(isZeroOp(opl)){
+     if(isZeroOp(opr))
+       return opr; // return a zero op if both are zero
+     if(isTpetrar){ // if other op is tpetra, replace this with a zero crs matrix
```

Comment:
Is there a reason to replace it with a TpetraOp, instead of just a ZeroLinearOp?

Reply:
I want to be able to call the Tpetra::MatrixMatrix::add later to create a new CrsMatrix and wrap it up to be consistent with the case where two Tpetra Crs matrices are input. I can't do that with a ZeroLinearOp.